pallets/msa/src/lib.rs

```diff
@@ -196,11 +218,16 @@ pub mod pallet {
 	/// - Key: Provider MSA Id
 	/// - Value: [`ProviderRegistryEntry`](common_primitives::msa::ProviderRegistryEntry)
 	#[pallet::storage]
-	pub type ProviderToRegistryEntry<T: Config> = StorageMap<
+	pub type ProviderToRegistryEntryV2<T: Config> = StorageMap<
```

Comment:
Now that I have more context surrounding migrations, is there a reason we're allocating a completely new storage here, versus migrating the existing storage in place?

Using `.drain()` + `.put()` costs 1 read + 2 writes per record migrated.
Using `.read()` old and `.insert()` new to the same storage map costs 1 read + 1 write.

Not a blocker, since mainnet only has a few providers and I guess we don't care so much about cost on Testnet, but just curious.

Reply:
I agree, thinking from that perspective definitely not a big migration however I would be open to following any standard practice 💯